src/Policy.ts

```diff
@@ -141,16 +161,50 @@ export class Policy<ReturnType> {
     predicate?: (error: ReturnType) => boolean,
   ) {
     const filter = typeFilter(cls, predicate);
-    return new Policy<ReturnType>(this.errorFilter, r => this.resultFilter(r) || filter(r));
+    return new Policy<ReturnType>({
+      ...this.options,
+      resultFilter: r => this.options.resultFilter(r) || filter(r),
+    });
   }
 
   /**
    * Returns a retry policy builder.
    */
   public retry() {
     return new RetryPolicy({
-      errorFilter: this.errorFilter,
-      resultFilter: this.resultFilter,
+      errorFilter: this.options.errorFilter,
+      resultFilter: this.options.resultFilter,
+    });
+  }
+
+  /**
+   * Returns a circuit breaker for the policy. **Important**: you should share
+   * your circuit breaker between executions of whatever function you're
+   * wrapping for it to function!
+   *
+   * ```ts
+   * import { SamplingBreaker, Policy } from 'cockatiel';
+   *
+   * // Break if more than 20% of requests fail in a 30 second time window:
+   * const breaker = Policy
+   *  .handleAll()
+   *  .circuitBreaker(new SamplingBreaker(0.2, 30 * 1000));
+   *
+   * export function handleRequest() {
+   *   return breaker.execute(() => getInfoFromDatabase());
+   * }
+   * ```
+   *
+   * @param breaker -- The circuit breaker to use. This package exports
+   * ConsecutiveBreaker and SamplingBreakers for you to use.
+   * @param halfOpenAfter -- Time after failures to try to open the circuit
+   * breaker again. Defaults to 10 seconds.
+   */
+  public circuitBreaker(breaker: IBreaker, halfOpenAfter = 10 * 1000) {
```

Comment:
While 10s might seem sensible to some applications, I'd recommend making this parameter required so people don't miss it.

Reply:
yep good call